src/plone/api/env.py

```diff
@@ -201,6 +202,17 @@ def test_mode():
     return IS_TEST
 
 
+def read_only_mode():
+    """Returns True if you are running the zope instance on an read only ZODB.
+
+    :Example: :ref:`env_read_only_mode_example`
+    """
+    conn = Globals.DB.open()
```

Comment:
That is the problem, the r/o configuration could change for a ZEO-Client at every request. See ZRS or RelStorage options in plone.recipe.zopeinstance:

    zeo-address = ${ZRS-Master} ${ZRS-Replica1} ... ${ZRS-ReplicaN} 

All ZRS-Replica are read-only by default, but ZRS-Master is writeable. 
ZRS-Master goes down, e.g. OS reboot, and ZRS-Replica takes over, at this moment the runtime config switches from read_only_mode = False to read_only_mode = True and back.